libdnf/rpm/repo_impl.hpp

```diff
@@ -92,8 +80,62 @@ class Key {
     long int timestamp;
 };
 
+// Information about attached libsolv repository
+class LibsolvRepoExt {
+public:
+    enum class DataType { FILENAMES, PRESTO, UPDATEINFO, OTHER };
+    enum class DataState { NEW, LOADED_FETCH, LOADED_CACHE, WRITTEN };
+
+    Id get_data_id(DataType which) const noexcept;
+    void set_data_id(DataType which, Id repodata) noexcept;
+    void set_data_state(DataType which, DataState state) noexcept;
+
+    // Returns "true" when all solvables in the repository are stored contiguously -> No interleaving
+    // with solvables from other repositories.
+    // Complexity: Linear to the current number of solvables in  repository
+    bool is_one_piece() const;
+
+    // Internalize repository if needed.
+    void internalize();
+
+    LibsolvRepo * repo{nullptr};
+
+    // Checksum of data in .solv file. Used for validity check of .solvx files.
+    unsigned char checksum[CHKSUM_BYTES];
+
+    DataState state_main{DataState::NEW};
```

Comment:
This construct is able to track only states of certain type of data. What about to store the information in map? Absent key could mean not requested.